aspnetcore/toc.yml

```diff
@@ -0,0 +1,807 @@
+    - name: Model validation
+      href: uid:mvc/models/validation
+    - name: Razor syntax
+      href: uid:mvc/views/razor
+    - name: Advanced
+      items: 
+        - name: View components
+          href: uid:mvc/views/view-components
+        - name: View compilation
+          href: uid:mvc/views/view-compilation
+        - name: App model
+          href: uid:mvc/controllers/application-model
+        - name: Filters
+          href: uid:mvc/controllers/filters
+        - name: Areas
+          href: uid:mvc/controllers/areas
+        - name: App parts
+          href: uid:mvc/extensibility/app-parts
+        - name: Custom model binding
+          href: uid:mvc/advanced/custom-model-binding
+        - name: Compatibility version
+          href: uid:mvc/compatibility-version
+- name: Web API apps
+  items: 
+    - name: Overview
+      href: uid:web-api/index
+    - name: Tutorials
+      items: 
+        - name: Create a web API
+          href: uid:tutorials/first-web-api
+        - name: Web API with MongoDB
+          href: uid:tutorials/first-mongo-app
+    - name: Swagger / OpenAPI
+      items: 
+        - name: Overview
+          href: uid:tutorials/web-api-help-pages-using-swagger
+        - name: Get started with Swashbuckle
+          href: uid:tutorials/get-started-with-swashbuckle
+        - name: Get started with NSwag
+          href: uid:tutorials/get-started-with-nswag
+    - name: Action return types
+      href: uid:web-api/action-return-types
+    - name: Format response data
+      href: uid:web-api/advanced/formatting
+    - name: Custom formatters
+      href: uid:web-api/advanced/custom-formatters
+    - name: Analyzers
+      href: uid:web-api/advanced/analyzers
+    - name: Conventions
+      href: uid:web-api/advanced/conventions
+- name: Real-time apps
+  items: 
+    - name: SignalR overview
+      href: uid:signalr/introduction
+    - name: Supported platforms
+      href: uid:signalr/supported-platforms
+    - name: Tutorials
+      items: 
+        - name: SignalR with JavaScript
+          href: uid:tutorials/signalr
+        - name: SignalR with TypeScript
+          href: uid:tutorials/signalr-typescript-webpack
+    - name: Samples
+      href: https://github.com/aspnet/SignalR-samples
+    - name: Server concepts
+      items: 
+        - name: Hubs
+          href: uid:signalr/hubs
+        - name: HubContext
+          href: uid:signalr/hubcontext
+        - name: Users and groups
+          href: uid:signalr/groups
+        - name: Publish to Azure
+          href: uid:signalr/publish-to-azure-web-app
+        - name: API design considerations
+          href: uid:signalr/api-design
+    - name: Clients
+      items: 
+        - name: .NET client
+          href: uid:signalr/dotnet-client
+        - name: .NET API reference
+          href: /dotnet/api/microsoft.aspnetcore.signalr.client
+        - name: Java client
+          href: uid:signalr/java-client
+        - name: Java API reference
```

Comment:
The Java API is available only in ASP.NET Core 2.2 and above. Is there a way to toggle visibility accordingly in this YAML-based ToC?

Reply:
So far as I know that only works by setting moniker range in article metadata, i.e., it's not a Markdown vs. YAML difference.